main.tf

```diff
@@ -1,9 +1,65 @@
 terraform {
-  required_version = ">= 1.3.0"
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.0"
+      version = "~> 5.0"
     }
+    cloudflare = {
+      source  = "cloudflare/cloudflare"
+      version = "~> 5.0"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 3.0"
+    }
+  }
+
+
+  required_version = ">= 1.11"
+}
+
+module "acm" {
+  source  = "terraform-aws-modules/acm/aws"
+  version = "5.1.1"
+
+  domain_name               = local.domains[0]
+  subject_alternative_names = slice(local.domains, 1, length(local.domains))
+
+  validation_method = "DNS"
+
+  create_route53_records = false
+  validate_certificate   = false
+}
```

Comment:
this needs to be more flexible, I would even go as far as an input var.acm type any. and document how to input the data.

this would error if you have only 1 domain